src/types/index.ts

```diff
@@ -135,4 +135,11 @@ export type ReduxAction = {
     payload?: any
     error?: boolean
     meta?: any
-}
\ No newline at end of file
+}
+
+export interface DeferredPromise<T> {
+    promise: Promise<T>;
+    resolve: T;
```

Comment:
I think it should be `(arg: T) => void` for resolve and `(err: Error) => void` for reject.